### ISSUE TITLE
Lambda async functions with await response

### ISSUE THREAD

**Issue by wesklei** [honeybadger-io/honeybadger-node#112]
Hi!

I'm trying to add honeybadger  in my serverless project and ended with an issue related to async lambda functions using RequestResponse invoke.  If I invoke my function using the code bellow the function hangs forever, it runs successfully but nothing is returned and the request keeps awaiting.
```
let params = {
        FunctionName: functionName,
        InvocationType: "RequestResponse",
        ....
    };
...
lambda.invoke(params);
...
```

My function declaration follow this pattern
```
const handler = async (params) => {
  ....
}

export default Honeybadger.lambdaHandler(handler);
```

I didn't found any example for request response, can you help me understand why my async function isn't working?

thanks!

ps: All my other functions that use Event invoke works.

**Comment by joshuap** [comment]
@wesklei I'm not sure off the top of my head why that would be failing. To be clear, the same function works with Event, but doesn't work with `RequestResponse`? Could you link any packages you are using to call the invoke API? I'll see if I can reproduce.

**Comment by wesklei** [comment]
thanks @joshuap 

I haven't tried the same function using Event yet. I have 6 functions that uses Event and work with Honeybadger, and I have 2 functions that use RequestResponse and don't

After I opened the ticket I've noticed the same with another case using Step Functions with Map. Perhaps the issue is related, since the map will wait for the step completion (probably behind has some sort of request response). In that case I have the following step function config

```
stepFunctions:
  validate: true
  stateMachines:
    cacheMapMachine:
      type: EXPRESS
      name: cacheMapMachine
      definition:
        StartAt: mapped_cache_generator_task
        States:
          mapped_cache_generator_task:
            Type: Map
            Next: EndState
            ItemsPath: "$.queueIds"
            Parameters:
              "queueIds.$": "$$.Map.Item.Value"
            Iterator:
              StartAt: CacheGeneratorMapTask
              States:
                CacheGeneratorMapTask:
                  Type: Task
                  Resource:
                    Fn::GetAtt: [CacheDashgenerateLambdaFunction, Arn]
                  End: true
          EndState:
            Type: Task
            Resource:
              Fn::GetAtt: [PreDashcacheDashlooperLambdaFunction, Arn]
            End: true
```
What happens is that the CacheDashgenerateLambdaFunction complete it's execution and instead of finish and invoke PreDashcacheDashlooperLambdaFunction, it hangs forever


Right now I overcome all this by wrapping my handlers with try catch and manually invoking Honeybadger.notify, not great but it works

**Comment by wesklei** [comment]
Attaching the code I used to invoke https://gist.github.com/wesklei/bd5a89add53acf4986a29630531b6e0f

**Comment by wesklei** [comment]
Also, including the rails code I use https://gist.github.com/wesklei/7e349088c09560955b52a5c987847c32
hope this helps

**Comment by joshuap** [comment]
Thanks, this is helpful. I'll take a look!

**Comment by joshuap** [comment]
@wesklei I'm testing an async lambda function:

https://github.com/honeybadger-io/serverless-quickstart/blob/honeybadger-node/issues/112/functions/greet.js

here's the code I'm using to invoke it, based on your example:

https://github.com/honeybadger-io/serverless-quickstart/blob/honeybadger-node/issues/112/invoke.rb

The function returns successfully for me. Here's a function that fails:

https://github.com/honeybadger-io/serverless-quickstart/blob/honeybadger-node/issues/112/functions/fail.js

That function reports an error to Honeybadger and also returns succesfully.

Any ideas what could be different between mine and yours?

**Comment by wesklei** [comment]
thanks @joshuap 👏 
I'll check what is different

**Comment by wesklei** [comment]
hi @joshuap 
Thanks for sharing that code, it helped me a lot to investigate!

After checking your code the only difference was me using serverless framework, specially serverless-offline plugin for local testing. When I was testing I started implementing locally using serverless-offline like I always do. The issue seems to be in serverless-offline or something else inside serverless framework. 

When I tested remotely, deploying my functions, it started working 🥳 . There was no issue in production, sorry! it was just locally 

Since I don't need Honeybadger locally, I will simply rewrite my code to export the handlers with lambdaHandler only on production


Thanks a lot for your help!
